invokeai/app/util/save_thumbnail.py

```diff
@@ -0,0 +1,28 @@
+import os
+from PIL import Image
+
+
+def save_thumbnail(
+    image: Image.Image,
+    filename: str,
+    path: str,
+    size: int = 256,
+) -> str:
+    """
+    Saves a thumbnail of an image, returning its path.
+    """
+    base_filename = os.path.splitext(filename)[0]
+    thumbnail_path = os.path.join(path, base_filename + ".webp")
+
+    if os.path.exists(thumbnail_path):
+        return thumbnail_path
+
+    thumbnail_width = size
+    thumbnail_height = round(size * (image.height / image.width))
+
+    image_copy = image.copy()
+    image_copy.thumbnail(size=(thumbnail_width, thumbnail_height))
```

Comment:
You don't need to pre-calculate width/height for the PIL thumbnail function. You just pass it the maximum size (e.g. 128x128) and it'll calculate the thumbnail size appropriately to maintain the aspect ratio.